File: ios/StockPro/Data/DTOs/LocationDTOs.swift
import Foundation

struct LocationDTO: Decodable, Identifiable {
    let id: Int
    let warehouse_id: Int
    let name: String
    let type: String?
    let capacity: Int?
    let warehouse_name: String?
    let created_at: String?

    func toItem() -> LocationListItem {
        LocationListItem(
            id: id,
            warehouseId: warehouse_id,
            name: name,
            type: type ?? "rack",
            capacity: capacity,
            warehouseName: warehouse_name ?? ""
        )
    }
}

struct LocationCreateRequest: Encodable {
    let name: String
    let warehouse_id: Int
    let type: String?
    let capacity: Int?
}

struct LocationUpdateRequest: Encodable {
    let name: String
    let warehouse_id: Int
    let type: String?
    let capacity: Int?
}

struct LocationListItem: Identifiable {
    let id: Int
    let warehouseId: Int
    let name: String
    let type: String
    let capacity: Int?
    let warehouseName: String

    var typeLabel: String {
        switch type {
        case "rack": "Rayon"
        case "shelf": "Étagère"
        case "zone": "Zone"
        default: type
        }
    }
}

File: App2/core/data/src/main/java/com/app2/core/data/remote/dto/LocationDTOs.kt
package com.app2.core.data.remote.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class LocationDTO(
    val id: Int,
    @SerialName("warehouse_id") val warehouseId: Int,
    val name: String,
    val type: String? = "rack",
    val capacity: Int? = null,
    @SerialName("created_at") val createdAt: String? = null
)

@Serializable
data class LocationCreateRequest(
    @SerialName("warehouse_id") val warehouseId: Int,
    val name: String,
    val type: String? = null,
    val capacity: Int? = null
)

@Serializable
data class LocationUpdateRequest(
    val name: String? = null,
    val type: String? = null,
    val capacity: Int? = null
)
